diff -rupN original/CMakeLists.txt patch/CMakeLists.txt
--- original/CMakeLists.txt	2018-02-26 17:29:05.000000000 -0800
+++ patch/CMakeLists.txt	2018-02-26 17:27:36.000000000 -0800
@@ -893,9 +893,9 @@ if(MITK_USE_QT)
     ERROR_VARIABLE _error
   )
   string(STRIP "${QT_BINARY_DIR}" QT_BINARY_DIR)
-  if(_result OR NOT EXISTS "${QT_BINARY_DIR}")
-    message(FATAL_ERROR "Could not determine Qt binary directory: ${_result} ${QT_BINARY_DIR} ${_error}")
-  endif()
+  #if(_result OR NOT EXISTS "${QT_BINARY_DIR}")
+  #  message(FATAL_ERROR "Could not determine Qt binary directory: ${_result} ${QT_BINARY_DIR} ${_error}")
+  #endif()
 
   find_program(QT_HELPGENERATOR_EXECUTABLE
     NAMES qhelpgenerator qhelpgenerator-qt5 qhelpgenerator5
