diff -aur --new-file -x .git mitk-2018.04.2/CMakeExternals/DCMTK.cmake mitk-2018.04.2-modified/CMakeExternals/DCMTK.cmake
--- mitk-2018.04.2/CMakeExternals/DCMTK.cmake	2019-06-30 21:12:44.000000000 -0700
+++ mitk-2018.04.2-modified/CMakeExternals/DCMTK.cmake	2019-06-30 21:14:03.000000000 -0700
@@ -36,6 +36,7 @@
          ${additional_args}
          "-DCMAKE_CXX_FLAGS:STRING=${CMAKE_CXX_FLAGS} ${DCMTK_CXX_FLAGS}"
          "-DCMAKE_C_FLAGS:STRING=${CMAKE_C_FLAGS} ${DCMTK_C_FLAGS}"
+	 -DDCMTK_ENABLE_BUILTIN_DICTIONARY:BOOL=ON
          -DDCMTK_ENABLE_CXX11:BOOL=ON
          -DDCMTK_ENABLE_STL:BOOL=ON
          -DDCMTK_WITH_DOXYGEN:BOOL=OFF
