diff -aur --new-file -x .git mitk-2018-04-alpha/CMake/mitkFunctionGetMSVCVersion.cmake mitk-2018-04-fixed/CMake/mitkFunctionGetMSVCVersion.cmake
--- mitk-2018-04-alpha/CMake/mitkFunctionGetMSVCVersion.cmake	2018-05-28 17:40:01.000000000 -0700
+++ mitk-2018-04-fixed/CMake/mitkFunctionGetMSVCVersion.cmake	2019-07-20 10:35:10.525736600 -0700
@@ -39,6 +39,10 @@
     set(VISUAL_STUDIO_VERSION_MAJOR "14" PARENT_SCOPE)
     set(VISUAL_STUDIO_VERSION_MINOR "3" PARENT_SCOPE)
     set(VISUAL_STUDIO_PRODUCT_NAME "Visual Studio 2017" PARENT_SCOPE)
+  elseif(MSVC_VERSION EQUAL 1916)
+    set(VISUAL_STUDIO_VERSION_MAJOR "14" PARENT_SCOPE)
+    set(VISUAL_STUDIO_VERSION_MINOR "3" PARENT_SCOPE)
+    set(VISUAL_STUDIO_PRODUCT_NAME "Visual Studio 2017" PARENT_SCOPE)
   else()
     message(WARNING "Unknown Visual Studio version ${MSVC_VERSION}. Please update CMake/mitkFunctionGetMSVCVersion.cmake.")
   endif()
